### PR TITLE
Adds analysis and layer interaction

### PR DESCRIPTION
**Pull request by javierarce**
This PR:

- [x] Allows drag & drop capabilities for analysis and layers.
- [x] Add basic styles for the analysis and layers.

The actual creation of the new analysis and layer will be resolved once we have the required functionalities, right now the result of the interaction appears in the console: 

- [layer creation placeholder](https://github.com/CartoDB/cartodb/pull/6923/files#diff-a329c4764c9d4f367b7626d084bb86fdR54)
- [analysis creation placeholder](https://github.com/CartoDB/cartodb/pull/6923/files#diff-a6465b31555b635aaa249639f4ce69a5R79)

Original issue: #6784

analysis_drag_drop
layer_drag_and_drop

### CODE REVIEW

**Comment by javierarce** [comment]
CR: @viddo 
CC: @xavijam && @piensaenpixel

File: lib/assets/javascripts/cartodb3/editor/layers/analysis-views/composite-layer-analysis-view.js [hunk 1]
```diff
@@ -1,4 +1,8 @@
+var cdb = require('cartodb-deep-insights.js');
 var template = require('./composite-layer-analysis-view.tpl');
+var LayerAnalysisDraggableHelperView = require('../layer-analysis-draggable-helper-view');
+
+require('jquery-ui/draggable');
```
Comment (viddo) [inline lib/assets/javascripts/cartodb3/editor/layers/analysis-views/composite-layer-analysis-view.js]: Why is the draggable lib required here? Don't see any direct uses, probably would be better to move it to where it's applied.

File: lib/assets/javascripts/cartodb3/editor/layers/analysis-views/ref-layer-analysis-view.js [hunk 2]
```diff
@@ -16,6 +25,18 @@ module.exports = cdb.core.View.extend({
...
+    });
+
+    this.addView(this.draggableHelperView);
+    this.draggableHelperView.bind('dropped', this._onDropped, this);
+
     return this;
+  },
+
+  _onDropped: function () {
+    console.log('TODO: creates a new layer from ' + this.model.toJSON()); // TODO: replace with actual layer generation
```
Comment (viddo) [inline lib/assets/javascripts/cartodb3/editor/layers/analysis-views/ref-layer-analysis-view.js]: Should one be able to drag a node reference?

File: lib/assets/javascripts/cartodb3/editor/layers/analysis-views/composite-layer-analysis-view.js (hunk 1 above)
Comment (viddo) [inline lib/assets/javascripts/cartodb3/editor/layers/analysis-views/composite-layer-analysis-view.js]: Isn't it sufficient with the [require in the helper](https://github.com/CartoDB/cartodb/pull/6923/files#diff-f1fe5927571d83445e20c0301e5bc308R3) already?

File: lib/assets/javascripts/cartodb3/editor/layers/layer-analysis-view-helper.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+ * @param {Object} layerAnalysisView
+ * @return {Object} a string representing the HTML code of the analysis view helper
+ */
+LayerAnalysisViewHelper.prototype.createHelper = function (layerAnalysisView) {
+  return '<li class="' + layerAnalysisView.className + '">' + layerAnalysisView.render().$el.html() + '</li>';
+};
+
+module.exports = LayerAnalysisViewHelper;
```
Comment (viddo) [inline lib/assets/javascripts/cartodb3/editor/layers/layer-analysis-view-helper.js]: I don't see clear value of having this separate (helper's helper), why not just inline this + use standard .tpl template in the draggable helper?

File: lib/assets/test/spec/cartodb3/editor/layers/layer-view.spec.js [hunk 4]
```diff
@@ -44,7 +44,25 @@ describe('editor/layers/layer-view', function () {
...
+    expect(this.newAnalysisViewsSpy.calls.argsFor(0)[0]).toEqual(this.view.$('.js-analyses'));
     expect(this.newAnalysisViewsSpy.calls.argsFor(0)[1]).toEqual(this.model);
   });
+
+  describe('draggable', function () {
+    it('should be initialized when view is rendered', function () {
+      spyOn(this.view, '_initDraggable').and.callThrough();
+      this.view.render();
+      expect(this.view._initDraggable).toHaveBeenCalled();
+      expect(this.view.$el.data('ui-draggable')).not.toBeUndefined();
```
Comment (viddo) [inline lib/assets/test/spec/cartodb3/editor/layers/layer-view.spec.js]: `.toBeDefined()`, and the spec should not test private stuff (`_initDraggable`) really, just assert this expected side-effect should be sufficient.

File: package.json [hunk 5]
```diff
@@ -53,7 +53,7 @@
     "grunt-postcss": "0.5.5",
     "grunt-timer": "0.3.3",
     "jquery": "2.1.4",
-    "jquery-ui": "^1.10.5",
+    "jquery-ui":"^1.10.5",
```
Comment (viddo) [inline package.json]: don't see any changes to the shrinkwrap, forgot to run `npm shrinkwrap --dev`?

**Comment by viddo** [comment]
Some minor comments, :+1: 

And let's tackle the side-effects (layer/node creation) in separate PRs then. :+1:

File: package.json (hunk 5 above)
Comment (javierarce) [inline package.json]: Fixed.

File: lib/assets/javascripts/cartodb3/editor/layers/analysis-views/ref-layer-analysis-view.js (hunk 2 above)
Comment (javierarce) [inline lib/assets/javascripts/cartodb3/editor/layers/analysis-views/ref-layer-analysis-view.js]: Yes, in order to create a new analysis layer.

**Comment by javierarce** [comment]
Merging pull request since the frontend tests passed.

**Pull request merged**